### ISSUE TITLE
Creating a conv2d LoRA layer always sets the bias, even if original layer had none

### ISSUE THREAD

**Issue by BenjaminBossan** [huggingface/peft#1013]
### System Info

-

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder
- [ ] My own task or dataset (give details below)

### Reproduction

Refer to this discussion: https://github.com/huggingface/peft/discussions/697#discussioncomment-7130768

### Expected behavior

When a `Conv2d` LoRA layer is created from a normal `Conv2d` layer, it will always have a bias, even if the original layer didn't have a bias. This can lead to errors down the line, as the colab notebook in the linked discussion shows.

Possible, this also affects other LoRA layer types and even other methods.

**Comment by BenjaminBossan** [comment]
@Nidhogg-lyz I could not reproduce your initial issue anymore (from [this notebook](https://colab.research.google.com/drive/1ZmOadYHMecEXxz-K_7p3JxSNhqdJ8Nzz?usp=sharing)). Could you please try again with the latest version of PEFT?

For completeness: `torch==2.1.0`, `torchvision==0.15.2+cu117`.

**Comment by Nidhogg-lyz** [comment]
> @Nidhogg-lyz I could not reproduce your initial issue anymore (from [this notebook](https://colab.research.google.com/drive/1ZmOadYHMecEXxz-K_7p3JxSNhqdJ8Nzz?usp=sharing)). Could you please try again with the latest version of PEFT?
> 
> For completeness: `torch==2.1.0`, `torchvision==0.15.2+cu117`.

@BenjaminBossan 
Thanks for the help on this issue. I ran the notebook again and this issue still exists. The package versions are `torch==2.0.1+cu118`, `torchvision==0.15.2+cu118`, and `peft==0.5.0`.

**Comment by BenjaminBossan** [comment]
I could confirm that the bug exists for `peft==0.5.0` but no longer with the latest PEFT from `main`. I'm not sure which commit fixed the issue, but it seems it's gone.

**Comment by Nidhogg-lyz** [comment]
I'll install it from the source to get the latest version. Thanks a lot!

**Comment by Nidhogg-lyz** [comment]
@BenjaminBossan 
That does work for me. Thx again!

**Comment by rylai111** [comment]
@BenjaminBossan 
I am really confused why __init__ in conv2d refuses to add the bias key, for example

bias = kwargs.pop("bias", True)
nn.Conv2d.__init__(self, in_channels, out_channels, kernel_size, stride, padding, bias=bias)

Is there any purpose of writing like this

**Comment by BenjaminBossan** [comment]
I haven't written the original code, so can't say what the intent was. But note that when we create a LoRA layer, we override the weight and bias with those from the original layer anyway, so it doesn't really matter if we init it with bias or not.

**Comment by rylai111** [comment]
> I haven't written the original code, so can't say what the intent was. But note that when we create a LoRA layer, we override the weight and bias with those from the original layer anyway, so it doesn't really matter if we init it with bias or not.
@BenjaminBossan When i print the intermediate result in the `forward` function, the bias with initialization joins computation. But finally the base model and the lora model gets the same result. (i 've used deepcopy)
